src/main/java/seedu/address/model/item/ItemDate.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.item;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class ItemDate {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Dates must follow the format yyyy-mm-dd or dd-mm-yyyy.";
+
+    // TODO: Handle validation
+    public static final String VALIDATION_REGEX = "yyyy-MM-dd";
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(VALIDATION_REGEX);
+
+    public static final String DATE_OUTPUT_PATTERN = "dd/MM/yyyy";
+
+    public final LocalDate itemDate;
+
+    public ItemDate(String unparsedDate) {
+        requireNonNull(unparsedDate);
+         checkArgument(isValidItemDatetime(unparsedDate), MESSAGE_CONSTRAINTS);
+        this.itemDate = getDatetimeFormat(unparsedDate); // TODO: Replace with parser
+    }
+
+    public static boolean isValidItemDatetime(String unparsedDatetime) {
+        DateValidator validator = new DateValidator(formatter);
+        return validator.isValid(unparsedDatetime);
+    }
+
+    public LocalDate getDatetimeFormat(String unparsedDatetime) {
+        return LocalDate.parse(unparsedDatetime, formatter);
```

Comment:
Do we need to assert that date time is parsable / do more checks here before we attempt to parse it?

Reply:
Hmm, I am just wondering if I should remove this function altogether, since validation would be then performed twice (once in the constructor, once in here). 

I am thinking in the future, if we need a utility function (or sth similar) to parse strings to datetime objects, they can just instantiate it using the constructor.